components/ReadingChallenge/ReadingChallenge.js

```diff
@@ -0,0 +1,192 @@
+import styled from "styled-components";
+import { useState } from "react";
+import useLocalStorageState from "use-local-storage-state";
+import ChallengeBookList from "../ChallengeBookList/ChallengeBookList";
+
+export default function ReadingChallenge({
+  books,
+  booksInfo,
+  handleToggleAlreadyRead,
+}) {
+  const [modalState, setModalState] = useState({
+    isOpen: false,
+    isCancelled: false,
+  });
+
+  function openModal() {
+    setModalState({ isOpen: true, isSaved: false });
+  }
+
+  function closeModal() {
+    setModalState({ isOpen: false, isSaved: false });
+  }
+  const [challenge, setChallenge] = useLocalStorageState("challenge", {
+    defaultValue: {},
+  });
+
+  const readBooks = books.filter((book) =>
+    booksInfo.find(
+      (bookInfo) => bookInfo.id === book.id && bookInfo.isAlreadyRead
+    )
+  );
+
+  const [progress, setProgress] = useLocalStorageState("progress", {
+    defaultValue: 0,
+  });
+
+  function updateProgress() {
+    if (challenge.type === "books") {
+      setProgress(readBooks.length);
+    } else if (challenge.type === "pages") {
+      const pagesArray = readBooks.map((readBook) => {
+        return readBook.pages;
+      });
+      const pagesSum = pagesArray.reduce((a, b) => a + b);
+      setProgress(pagesSum);
+    }
+  }
+  updateProgress();
```

Comment:
Currently, `updateProgress` is called directly in the component body, which means it runs on every re-render. This could be inefficient, especially if `readBooks` or `challenge.type` changes infrequently.

To be able to offer you an alternative besides a `useEffect` hook, I would have to dig a little deeper into the project. Maybe this is a case for a refactor note. 🚧 

It's okay for now.